doc/user/content/sql/create-source/kafka.md

```diff
@@ -299,6 +299,18 @@ As long as any offset continues increasing, Materialize is consuming data from
 the upstream Kafka broker. For more details on monitoring source ingestion
 progress and debugging related issues, see [Troubleshooting](/ops/troubleshooting/).
 
+#### Progress details
+
+The progress collection describes which Kafka messages are available in the
+source itself at a given time. Namely, all records in each reported partition
+with offsets up to but not including (i.e. `<`) the reported offset will be
+considered.
+
+This means that any queries that assess the source's progress collection will
+include all messages from the source whose offsets are described. You could, for
+instance, join a query of the source and a query of the progress collection to
```

Comment:
I think that will produce annoying results for sources whose progress has more than one row because the only possible join is a cross join. Instead, if someone wants to do a query and also know which offsets from the source are included in the results we should suggest a transaction:

```sql
BEGIN;
SELECT * FROM my_source_progress;
SELECT * FROM my_source; -- Or any other complicated query
COMMIT:
```

By running this in a transaction we guarantee that the observation of progress and query run at the same timestamp.